### ISSUE TITLE
Rust: don't show run button in build scripts

### ISSUE THREAD

**Issue by kzhui125** [zed-industries/zed#14720]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

<img width="418" alt="image" src="https://github.com/user-attachments/assets/5e4617b8-1917-41fb-bc41-f9ad2b56cead">

### Environment

Zed: v0.144.3 (Zed)
OS: macOS 14.5.0
Memory: 32 GiB
Architecture: aarch64

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>

**Comment by versecafe** [comment]
Every `fn main()` in rust is supposed to have the arrow with it, the file name shouldn't change that behaviour, what is your issue with it appearing in `build.rs`

**Comment by kzhui125** [comment]
this is what in vscode, `fn main` in build.rs don't have run button @versecafe 

<img width="322" alt="image" src="https://github.com/user-attachments/assets/2b351cbd-6a2b-40a8-affa-e958285312b1">

**Comment by kzhui125** [comment]
to clarify, not all build.rs files, only rust build scripts

https://doc.rust-lang.org/cargo/reference/build-scripts.html

**Comment by osiewicz** [comment]
I agree that we shouldn't show a run button in build scripts, but to just put it out there: it's a bit harder than just checking if a file is named `build.rs`, as the name of a build script can be arbitrary. We'd have to piggy-back out of output of `cargo metadata` for that functionality.

**Comment by kzhui125** [comment]
also files not in crates:

<img width="921" alt="image" src="https://github.com/user-attachments/assets/f0652491-9052-4d99-b1d5-2165dcb12595">

**Comment by versecafe** [comment]
If the rust analyzer isn't doing anything neither is the LSP and the LSP is where the hooks for the test functions go so that last example isn't configured correctly, also why is there an expectation that it shouldn't be there, is there any issue to offering the feature?

**Comment by kzhui125** [comment]
`run button` should only exists when it's runnable @versecafe

**Comment by versecafe** [comment]
Build scripts are runnable, you usually don't want to run them on their own but they should be runnable

**Comment by kzhui125** [comment]
build scripts are for `cargo build`, run them on their own are not supported now https://github.com/rust-lang/cargo/issues/7178

We use `run` to print output (for example println!)

but output is not supported by build scripts now. https://github.com/rust-lang/cargo/issues/985

**Comment by kzhui125** [comment]
Should `run` button be added for build scripts, click it doesn't work now(will be another bug).
